Section #4 - Capstone/simpsons.py

```diff
@@ -76,14 +79,38 @@
 train_gen = datagen.flow(x_train, y_train, batch_size=BATCH_SIZE)
 
 # Create our model (returns the compiled model)
-model = canaro.models.createSimpsonsModel(IMG_SIZE=IMG_SIZE, channels=channels, output_dim=len(characters), 
-                                         loss='binary_crossentropy', decay=1e-7, learning_rate=0.001, momentum=0.9,
-                                         nesterov=True)
+output_dim=10
+
+w, h = IMG_SIZE[:2]
+
+model = Sequential()
+model.add(Conv2D(32, (3, 3), activation='relu', padding='same', input_shape=(w, h,channels)))
+model.add(Conv2D(32, (3, 3), activation='relu'))
+model.add(MaxPooling2D(pool_size=(2, 2)))
+model.add(Dropout(0.2))
+
+model.add(Conv2D(64, (3, 3), padding='same', activation='relu'))
+model.add(Conv2D(64, (3, 3), activation='relu'))
+model.add(MaxPooling2D(pool_size=(2, 2)))
+model.add(Dropout(0.2))
+
+model.add(Conv2D(256, (3, 3), padding='same', activation='relu')) 
+model.add(Conv2D(256, (3, 3), activation='relu'))
+model.add(MaxPooling2D(pool_size=(2, 2)))
+model.add(Dropout(0.2))
+
+model.add(Flatten())
+model.add(Dropout(0.5))
+model.add(Dense(1024, activation='relu'))
+
+# Output Layer
+model.add(Dense(output_dim, activation='softmax'))
 
 model.summary()
 
 # Training the model
-
+optimizer = SGD(learning_rate=0.001, decay=1e-7, momentum=0.9, nesterov=True)
```

Comment:
in july 2024, this optimizer is not supported in Keras 3.
just change the parameter : decay ==> weight_decay and for importing use tensorflow.keras.optimizers instead of tensorflow.keras.optimizers.legacy